<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="node_name"     default="iri_demos_kinova"/>
  <arg name="output"        default="screen"/>
  <arg name="launch_prefix" default=""/>
  <arg name="config_file"   default="$(find iri_demos_kinova)/config/params.yaml"/>
  <!-- <arg name="topic_name"  default="new_topic_name"/> -->
  <arg name="diagonal_m" default="true"/>
  <arg name="garment_width" default="0.26"/>
  <arg name="garment_height" default="0.24"/>
  <!--<arg name="pre_grasp_corner" default="[0.550, 0.372, 0.055, 0.0, -125.5, 180.0]"/>-->
<!--  <arg name="pre_grasp_corner" default="[0.635, 0.17, 0.055, 0.0, -125.5, 180.0]"/> -->
  <arg name="pre_grasp_corner" default="[0.60, 0.17, 0.055, 0.0, -125.5, 180.0]"/>


  <node name="$(arg node_name)"
        pkg ="iri_demos_kinova"
        type="iri_demos_kinova"
        output="$(arg output)"
        launch-prefix="$(arg launch_prefix)">
	<rosparam file="$(arg config_file)" command="load"/>
	<param name="diagonal_move" value="$(arg diagonal_m)"/>
  <param name="garment_width" value="$(arg garment_width)"/>
  <param name="garment_height" value="$(arg garment_height)"/>
  <rosparam param="pre_grasp_corner" subst_value="True">$(arg pre_grasp_corner)</rosparam>
  <remap from="~/kinova_linear_move" to="/iri_kinova_linear_movement/kinova_linear_move"/>
    <!--<remap from="~/topic" to="$(arg topic_name)"/>-->
  </node>

</launch>
